fastmri/data/mri_data.py

```diff
@@ -358,3 +360,111 @@ def __getitem__(self, i: int):
             sample = self.transform(kspace, mask, target, attrs, fname.name, dataslice)
 
         return sample
+
+
+class AnnotatedSliceDataset(SliceDataset):
+    def __init__(
+            self,
+            root: Union[str, Path, os.PathLike],
+            challenge: str,
+            transform: Optional[Callable] = None,
+            use_dataset_cache: bool = False,
+            sample_rate: Optional[float] = None,
+            volume_sample_rate: Optional[float] = None,
+            dataset_cache_file: Union[str, Path, os.PathLike] = "dataset_cache.pkl",
+            num_cols: Optional[Tuple[int]] = None,
+            use_annotation: bool = False,
+            annotation_version: Optional[str] = None,
+            mri_type: Optional[str] = None,
+            multiple_annotation_policy: Optional[str] = None
+    ):
+        # subclass SliceDataset
+        SliceDataset.__init__(self, root, challenge, transform, use_dataset_cache, sample_rate, volume_sample_rate,
+                              dataset_cache_file, num_cols)
+
+        self.annotated_examples = []
+
+        if use_annotation is True:
+            # download csv file from github using git hash to find certain version
+            annotation_file_name = self.download_csv(annotation_version, mri_type)
```

Comment:
We should check if we already have the file before download.

Also, this looks like a bug as `mri_type` can be `None`. `mri_type` should not be optional I think.

Reply:
Done. 
Also, `multiple_annotation_policy` is required to be either "first", ”random“ or ”all".
For annotation_version, it's optional. If it's none, it will be latest version.